src/test/java/dog/pawbook/logic/commands/EnrolCommandTest.java

```diff
@@ -0,0 +1,152 @@
+
+        expectedModel.updateFilteredEntityList(new IdMatchPredicate(programIdSet));
+
+        EnrolCommand enrolCommand = new EnrolCommand(dogIdSet, programIdSet);
+        String expectedMessage = String.format(EnrolCommand.MESSAGE_SUCCESS_FORMAT, ID_TWO, ID_FIFTEEN);
+        assertCommandSuccess(enrolCommand, model, expectedMessage, expectedModel);
+    }
+
+    @Test
+    public void execute_enrolOneDogManyPrograms_successful() {
+        Set<Integer> dogIdSet = new HashSet<>();
+        Set<Integer> programIdSet = new HashSet<>();
+        dogIdSet.add(ID_TWO);
+        programIdSet.add(ID_FIFTEEN);
+        programIdSet.add(ID_SIXTEEN);
+
+        Program editedProgram1 = (Program) expectedModel.getEntity(15);
+        Program editedProgram2 = (Program) expectedModel.getEntity(16);
+
+        HashSet<Integer> updatedEnrolledDogs1 = new HashSet<>(editedProgram1.getDogIdSet());
+        updatedEnrolledDogs1.add(2);
+
+        expectedModel.setEntity(15, new Program(editedProgram1.getName(),
+                editedProgram1.getSessions(), editedProgram1.getTags(), updatedEnrolledDogs1));
+
+        HashSet<Integer> updatedEnrolledDogs2 = new HashSet<>(editedProgram1.getDogIdSet());
+        updatedEnrolledDogs2.add(2);
+
+        expectedModel.setEntity(16, new Program(editedProgram2.getName(),
+                editedProgram2.getSessions(), editedProgram2.getTags(), updatedEnrolledDogs2));
+
+        expectedModel.updateFilteredEntityList(new IdMatchPredicate(programIdSet));
+
+        EnrolCommand enrolCommand = new EnrolCommand(dogIdSet, programIdSet);
+        String expectedMessage = String.format(EnrolCommand.MESSAGE_SUCCESS_FORMAT, ID_TWO,
+                programIdSet.stream().map(String::valueOf).collect(Collectors.joining(", ")));
+
+        assertCommandSuccess(enrolCommand, model, expectedMessage, expectedModel);
+    }
+
+    @Test
+    public void execute_enrolManyDogsOneProgram_successful() {
+        Set<Integer> dogIdSet = new HashSet<>();
+        Set<Integer> programIdSet = new HashSet<>();
+        dogIdSet.add(ID_TWO);
+        dogIdSet.add(ID_FOUR);
+        dogIdSet.add(ID_SIX);
+        programIdSet.add(ID_FIFTEEN);
+
+        Program editedProgram = (Program) expectedModel.getEntity(15);
+
+        HashSet<Integer> updatedEnrolledDogs = new HashSet<>(editedProgram.getDogIdSet());
+        updatedEnrolledDogs.addAll(dogIdSet);
+
+        expectedModel.setEntity(15, new Program(editedProgram.getName(),
+                editedProgram.getSessions(), editedProgram.getTags(), updatedEnrolledDogs));
+
+        expectedModel.updateFilteredEntityList(new IdMatchPredicate(programIdSet));
+
+        EnrolCommand enrolCommand = new EnrolCommand(dogIdSet, programIdSet);
+        String expectedMessage = String.format(EnrolCommand.MESSAGE_SUCCESS_FORMAT,
+                dogIdSet.stream().map(String::valueOf).collect(Collectors.joining(", ")),
+                ID_FIFTEEN);
```

Comment:
Yay good use of streams, but it will be more readable if it is something like:
```
String expectedMessage = String.format(EnrolCommand.MESSAGE_SUCCESS_FORMAT,
        dogIdSet.stream()
                .map(String::valueOf)
                .collect(Collectors.joining(", ")),
        ID_FIFTEEN);
```

Of course, adjust the indentation as necessary.

Reply:
okie!